lib/router/src/lib.rs

```diff
@@ -71,17 +72,17 @@ where
 {
     recognize: Rec,
     make: Mk,
-    cache: Mutex<Cache<Rec::Target, Mk::Value>>,
+    cache: Mutex<Cache<Rec::Target, LoadShed<Mk::Value>>>,
 }
 
 enum State<Req, Svc>
 where
     Svc: svc::Service<Req>,
+    Svc::Error: Into<error::Error>,
 {
-    NotReady(Req, Svc),
-    Called(Svc::Future),
-    Error(error::Error),
-    Tmp,
+    Init(Option<Req>, LoadShed<Svc>),
```

Comment:
TIOLI: Does `State` need to know about the `LoadShed` middleware? It seems like it could've been left the same and `ResponseFuture` just be changed to have
```rust
state: State<Req, LoadShed<Svc>>,
```